### PR TITLE
Bump libgd to version 2.1.1

### PR DESCRIPTION
**Pull request by jimklimov**

### CODE REVIEW

File: components/library/gd2/gd2.p5m [hunk 1]
```diff
@@ -96,18 +96,17 @@ file path=usr/include/gd2/gdfontmb.h
...
+file path=usr/lib/$(MACH64)/libgd.so.3.0.0
 file path=usr/lib/$(MACH64)/llib-lgd.ln
-file path=usr/lib/libgd.so.2.0.0
+file path=usr/lib/libgd.so.3.0.0
 file llib-lgd path=usr/lib/llib-lgd
 file path=usr/lib/llib-lgd.ln
 
-file entities.html path=usr/share/doc/gd2/html/entities.html
-file index.html path=usr/share/doc/gd2/html/index.html
+file src/entities.html path=usr/share/doc/gd2/html/entities.html
```
Comment (pyhalov) [inline components/library/gd2/gd2.p5m:111]: Why should we deliver it?

File: components/library/gd2/gd2.p5m [hunk 2]
```diff
@@ -96,18 +96,17 @@ file path=usr/include/gd2/gdfontmb.h
 file path=usr/include/gd2/gdfonts.h
 file path=usr/include/gd2/gdfontt.h
 file path=usr/include/gd2/gdfx.h
-file path=usr/include/gd2/gdhelpers.h
-file path=usr/include/gd2/jisx0208.h
-file path=usr/include/gd2/wbmp.h
+file src/gdhelpers.h path=usr/include/gd2/gdhelpers.h
```
Comment (pyhalov) [inline components/library/gd2/gd2.p5m]: is it now not a part of API?

**Comment by pyhalov** [comment]
This is ABI-incompatible change. Do we really want this update? If we do, the following packages should be checked to build correctly after this update:

pkg:/diagnostic/mrtg
pkg:/image/gnuplot
pkg:/image/graphviz
pkg:/image/graphviz/graphviz-tcl
pkg:/web/analytics/webalizer
pkg:/web/php-5*/extension/php-gd

**Comment by pyhalov** [comment]
and also vlc from encumbered

**Comment by jimklimov** [comment]
"Need" - not so much, just thought we should promote our baseline. I'll see
if those other packages like this gd.

As for headers and the llib file - this is legacy from old version. If it
was delivered earlier - i chose to deliver it anew. Also the llib is
somehow used in the build and should stay in component rootdir (moving to
files/ breaks build).

## 

Typos courtesy of GMail on my Samsung Android
13.04.2016 6:19 пользователь "Alexander Pyhalov" notifications@github.com
написал:

**Comment by jimklimov** [comment]
Also to review: rrdtool (maybe more so than mrtg itself), maybe imagemagick

**Comment by jimklimov** [comment]
Hmmm... this sounds like a good idea, and it is disturbing that support is not detected:

```
checking if libraries can be versioned... no
configure: WARNING: *** You have not enabled versioned symbols.
```

Any ideas? :-)

**Comment by pyhalov** [comment]
look at the code, they can support only gnu ld

**Comment by jimklimov** [comment]
Installed my builds of libpng14 and libpng16, as well as older 1.0 and 1.2, and my build of libdg-2.1.1.

mrtg, webalizer, rrdtool - ok, but the binaries refer to both libpng14 and libpng16 (via "ldd" inspection), and new libgd.so.3 where used

graphviz got built, but without installed php - failed to build deliverables for php ;) yet, the binary files built also referenced both 14 and 16, and gd

imagemagick ok, but only linked with libpng14 - not gd, not 1.6

php and remake of graphivz hopefully will build overnight, no errors so far...

**Comment by jimklimov** [comment]
And what is special about gnu ld? :)
In linux at least, it is able to build versioned-API libraries...

**Comment by jimklimov** [comment]
php56 rebuilt ok, the gd.so extension refers to libpng16 only.
even with "php56" (only) installed, something is missing for graphviz to build php files, but until the "gmake package" step all is ok, and libs refer to png14 png16 gd.

**Comment by jimklimov** [comment]
vlc wants some more encumbered dependencies, so will take a while
but at least for "gmake publish" of those other packages, the updated LIBPNG and LIBGD posed no problem.

**Comment by pyhalov** [comment]
Likely you see dependencies on multiple libpng versions because A links with libpngV1 and shared library B is linked to libpngV2 and A is also linked to B.

**Comment by pyhalov** [comment]
gnu ld has different options to specify symbol versions

**Comment by jimklimov** [comment]
Is it a problem to tackle (and how?) Or can this stay as is (at least for
now, maybe somebody smarter about linking will later make a fix) - since we
deliver all versions of both libs and consumers anyway?..

## 

Typos courtesy of GMail on my Samsung Android
14.04.2016 9:58 пользователь "Alexander Pyhalov" notifications@github.com
написал:

**Comment by pyhalov** [comment]
Binaries linked to different libpng versions is a problem. Symbol versioning is not.

**Comment by jimklimov** [comment]
libpng14 seems to stem from at least fontconfig which is linked against it, and its pkg-config reports the settings for compilation... looking if rebuild would help...

UPDATE: libfreetype is the culprit

**Comment by jimklimov** [comment]
After discussion in IRC, here's an update to also build and provide the gdlib-2.0.35 (`*.so.2*` files only).

File: components/library/gd2/gd2.p5m (hunk 1 above)
Comment (jimklimov) [inline components/library/gd2/gd2.p5m:111]: No idea, this is part of original recipe. index is no longer present anywhere for libgd-2.1..

Comment (pyhalov) [inline components/library/gd2/gd2.p5m:111]: I wouldn't care about /usr/share/doc "stability"

**Comment by jimklimov** [comment]
Commits collapsed into one...

**Comment by jimklimov** [comment]
Does anything preclude merging this? ;)

**Comment by pyhalov** [comment]
libtool: compile:  /usr/gcc/4.9/bin/gcc -std=gnu99 -DHAVE_CONFIG_H -I. -I/export/home/alp/srcs/oi-userland/components/library/gd2/libgd-gd-2.1.1/src -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -m32 -O3 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -fvisibility=hidden -MT webpimg.lo -MD -MP -MF .deps/webpimg.Tpo -c webpimg.c  -fPIC -DPIC -o .libs/webpimg.o
libtool: compile:  /usr/gcc/4.9/bin/gcc -std=gnu99 -DHAVE_CONFIG_H -I. -I/export/home/alp/srcs/oi-userland/components/library/gd2/libgd-gd-2.1.1/src -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -m32 -O3 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -fvisibility=hidden -MT gd_color_match.lo -MD -MP -MF .deps/gd_color_match.Tpo -c gd_color_match.c  -fPIC -DPIC -o .libs/gd_color_match.o
libtool: compile:  /usr/gcc/4.9/bin/gcc -std=gnu99 -DHAVE_CONFIG_H -I. -I/export/home/alp/srcs/oi-userland/components/library/gd2/libgd-gd-2.1.1/src -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -m32 -O3 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -fvisibility=hidden -MT gd_bmp.lo -MD -MP -MF .deps/gd_bmp.Tpo -c gd_bmp.c  -fPIC -DPIC -o .libs/gd_bmp.o
libtool: compile:  /usr/gcc/4.9/bin/gcc -std=gnu99 -DHAVE_CONFIG_H -I. -I/export/home/alp/srcs/oi-userland/components/library/gd2/libgd-gd-2.1.1/src -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/harfbuzz -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -m32 -O3 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -fvisibility=hidden -MT gd_xbm.lo -MD -MP -MF .deps/gd_xbm.Tpo -c gd_xbm.c  -fPIC -DPIC -o .libs/gd_xbm.o
webpimg.c: In function 'VPXEncode':
webpimg.c:714:24: error: 'IMG_FMT_I420' undeclared (first use in this function)
     vpx_img_wrap(&img, IMG_FMT_I420,
                        ^
webpimg.c:714:24: note: each undeclared identifier is reported only once for each function it appears in
webpimg.c:716:16: error: 'PLANE_Y' undeclared (first use in this function)
     img.planes[PLANE_Y] = (uint8_)(Y);
                ^
webpimg.c:717:16: error: 'PLANE_U' undeclared (first use in this function)
     img.planes[PLANE_U] = (uint8_)(U);
                ^
webpimg.c:718:16: error: 'PLANE_V' undeclared (first use in this function)
     img.planes[PLANE_V] = (uint8*)(V);
                ^
Makefile:886: recipe for target 'webpimg.lo' failed

**Comment by jimklimov** [comment]
Can not reproduce the issue - `gmake clean publish` works for me, even after merge from upstream (just in case something became incompatible).

UPD: Retrying with explicitly disabled CCACHE... UPD2: still (re)builds from scratch OK...

**Comment by pyhalov** [comment]
do you run it on up-to-date system?

**Comment by jimklimov** [comment]
Interesting point - last updated about a week ago. Did you break much in
the meanwhile? ;*)

## 

Typos courtesy of GMail on my Samsung Android
25.05.2016 16:41 пользователь "Alexander Pyhalov" notifications@github.com
написал:

**Comment by jimklimov** [comment]
Updated hipster, git-pull-merged the userland recipes, now running the cleanest rebuild I can offer:

```
ENABLE_CCACHE=false gmake clean publish
```

**Comment by jimklimov** [comment]
Passed OK

**Comment by pyhalov** [comment]
http://trac.sagemath.org/ticket/18293 , the following patch https://raw.githubusercontent.com/PacBSD/abs/master/extra/gd/gd-2.1.1-libvpx-1.4.0.patch helps

Don't forget to add libvpx to REQUIRED_PACKAGES and see if delivered files have changed

**Comment by pyhalov** [comment]
otherwise LGTM. 
Affected components rebuild fine with updated gd2

**Comment by jimklimov** [comment]
Updated with patch and requirement for libvpx
BTW: Maybe go to update libvpx itself (yet another incompatible ABI bump) and/or add other code from http://www.webmproject.org/code/ later on soon?

**Pull request merged**